### PR TITLE
Rename Subsets Plugin

### PR DESCRIPTION
**Pull request by haticekaratay**
### Description

This pull request 
  - Restores the plugin name from `Subsets` back to `Subset Tools,` returning it to its original state.
  - Renames app bar `Subset Tools` to `SubsetAppBarWidget`

Fixes #<Issue Number>

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py
Comment (kecnry) [inline jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py:1]: can you please also change the path to this file (and class) so that it is `configs.default.plugins.subset_tools.subset_tools.SubsetTools`?

File: jdaviz/configs/default/plugins/subset_tools/subset_tools.py
Comment (kecnry) [inline jdaviz/configs/default/plugins/subset_tools/subset_tools.py:1]: and also the path here to something that clearly isn't the plugin?

**Review by kecnry — COMMENTED** [review]
Thanks (and sorry for the whiplash! 😉 )

The app-bar widget should also be removed from the API docs (currently appearing at https://jdaviz--3304.org.readthedocs.build/en/3304/reference/api_plugins.html#module-jdaviz.configs.default.plugins.subset_tools.subset_tools)

**Review by cshanahan1 — COMMENTED** [review]
Looks good! my personal opinion would be to rename configs/default/plugins/subset_tools and subset_tools.py to something more clear like subset_app_bar_widget

**Comment by haticekaratay** [comment]
> Looks good! my personal opinion would be to rename configs/default/plugins/subset_tools and subset_tools.py to something more clear like subset_app_bar_widget

That crossed my mind as well, but it felt quite long, so I renamed it to `subset_widget`. I'm happy to go with whatever the consensus is on this.

**Review by kecnry — COMMENTED** [review]
was git used to rename the files?  For some reason its showing as the old file entirely deleted and a new one created instead of just a rename (but maybe that is because of the path rename as well?)

**Comment by pllim** [comment]
> For some reason its showing as the old file entirely deleted and a new one created

Because the contents also changed (e.g., class name).

**Comment by kecnry** [comment]
Can we do that in two separate commits then to retain history?

**Comment by haticekaratay** [comment]
I have a plan to make it happen, and I am currently working on it.

**Comment by haticekaratay** [comment]
@kecnry 
I moved several files, such as `subset_plugin.vue`, `subset_plugin.py`, and others, into a new `subset_tools` submodule. However, Git isn’t fully preserving the history for these files in the new location, even though I used `git mv` and kept changes separate.
If contributors who made significant contributions to these files could rename and commit the path changes directly to my branch, it might preserve the history better. Feel free to commit to my branch or share suggestions here if anyone has a solution. Thanks!

**Review by kecnry — APPROVED** [review]
Thanks for trying!  Looks like it might not be possible with the directory rename, but git may still be able to follow this history with `--follow` (for future inspections).

**Comment by pllim** [comment]
You could try something like this (after merge) but not sure how well it works if you are moving things around. It was nice when we applied `black` to the entire code base.

https://github.com/astropy/astropy/blob/main/.git-blame-ignore-revs

**Comment by haticekaratay** [comment]
@kecnry Thank you for your approval. I have rebased to ensure the `readthedocs` build is successful. I will merge it once the build is completed.

**Review by cshanahan1 — APPROVED** [review]

**Pull request merged**
